src/nsolid/nsolid_bpf.h

```diff
@@ -46,10 +54,38 @@ inline void cap_free(cap_t) {}
 
 namespace node {
 namespace nsolid {
-// Define Linux-specific struct that works on all platforms
+
+enum class EbpfLoadStatus {
+  SUCCESS,
+  FILE_NOT_FOUND,
+  LOAD_ERROR,
+  NOT_ALLOWED,
+  UNKNOWN_ERROR,
+  NOT_SUPPORTED
+};
+
+class EbpfLoader {
+ public:
+  EbpfLoader();
+  ~EbpfLoader();
+
+  EbpfLoadStatus LoadProgram(const std::string& program_name);
+
+  // Clean up all loaded BPF objects
+  static void CleanupAllBpfObjects();
+
+ private:
+  const std::unordered_set<std::string> allowed_programs_;
+  const std::string ebpf_program_path_;
+
+  // Static vector to keep track of all loaded BPF objects
+#if HAVE_LIBBPF
+  static std::vector<struct ::bpf_object*> loaded_objects_;
+#endif
+};
```

Comment:
Why making `loaded_objects_` static (along with the `CleanupAllBpfObjects()` method?

Reply:
This way, we have full control of all loaded objects for a proper cleanup. Otherwise, on process shutdown, we'll need to get all loader instances and access their loaded objects (to simplify lifecycle).

Reply:
For example, if multiple Envs are created (worker threads), we would need to deal with loading ePBF programs twice. Adding it to node_main_instance (and having a static method), we ensure we just have one place with all loaded programs.

Reply:
Oh, right. I thought we had one EnvList per Env.